search/releases/7.1/index.adoc

```diff
@@ -0,0 +1,56 @@
+:awestruct-layout: project-releases-series
+:awestruct-project: search
+:awestruct-series_version: "7.1"
+:page-interpolate: true
+:latest-release-version: #{series(page, page.series_version).releases.first.version}
+:hsearch-doc-url-prefix: #{reference_doc(site.projects[page.project], series(page, page.series_version)).html_url}
+
+[[vector-search]]
+=== Vector search for the Lucene Backend
+
+Hibernate Search supports vector search in the Lucene backend as an incubating feature.
+Vector search provides the tools to build a semantic search for binary (images, audio or video) or text data.
+Note that there are no built-in models or high-level abstractions to convert binary/text data into vectors.
+Hibernate Search introduces a new field type -- `@VectorField` and a new predicate `knn`, so that the vectors can be indexed
+and then searched upon.
+
+Vector fields can work with vector data represented as `byte` or `float` arrays in the documents.
+Out of the box `byte[]` and `float[]` property types will work with the new field type. For any other entity property types,
+a custom binder should be implemented. Keep in mind that indexed vectors *must* be of the same length
+and that this length should be specified upfront for the schema to be created:
+
+[source, Java, indent=0, subs="+attributes"]
+----
+@Entity
+@Indexed
+public class Book {
+
+	@Id
+	private Integer id;
+
+	@VectorField(dimension = 512)
```

Comment:
I'd feel safer if you found an external source telling us that 512 dimensions make sense for image vectors :]

Reply:
seems safe enough 🙈: https://www.elastic.co/blog/implement-image-similarity-search-elastic will lead to:
```json
{
  "settings": {
    "index.refresh_interval": "5s",
    "number_of_shards": 1
  },
  "mappings": {
    "properties": {
      "image_embedding": {
        "type": "dense_vector",
        "dims": 512,
        "index": true,
        "similarity": "cosine"
      },
      "image_id": {
        "type": "keyword"
      },
      "image_name": {
        "type" : "keyword"
      },
      "relative_path" : {
        "type" : "keyword"
      },
      "exif" : {
        "properties" : {
          "location": {
            "type": "geo_point"
          },
          "date": {
            "type": "date"
          }
        }
      }
    }
  }
}
```

Reply:
Ok, thanks.

FYI @gsmet it seems I was wrong when I told you arrays used for vector search would necessarily use a low number of dimensions. Here it seems 512 dimensions (`array.length == 512`) is reasonable.
I did read somewhere that 3-4 dimensions were the norm, but I guess I shouldn't have trusted a single source.